posts/2022-01-31-configuring-ocp-for-user-namespace.md

```diff
@@ -0,0 +1,198 @@
+---
+title: Configuring ocp for user namespace
+published_date: "2022-01-31 11:45:58 +0000"
+layout: default.liquid
+is_draft: false
+---
+Preparing an openshift cluster for using user namespaces involve
+several steps and hands-over. To simplify the process we are using
+some configurations at freeipa-kustomize that make easier that task.
+
+**Pre-requisites**:
+
+- A 4.9+ OpenShift cluster.
+- You are logged in the cluster and you have cluster-admin privileges.
+- runc and cri-o rpm patched packages. Credits and thanks to [Fraser Tweedale](https://frasertweedale.github.io/blog-redhat/).
+  - [runc-1.0.3-992.rhaos4.10.el8.x86_64.rpm](https://ftweedal.fedorapeople.org/runc-1.0.3-992.rhaos4.10.el8.x86_64.rpm).
+  - [cri-o-1.23.0-990.rhaos4.10.git8c7713a.el8.x86_64.rpm](https://ftweedal.fedorapeople.org/cri-o-1.23.0-990.rhaos4.10.git8c7713a.el8.x86_64.rpm).
+
+> **Out of scope**: Build the runc and cri-o rpm packages.
+
+## Setting the configuration node
+
+- Clone freeipa-kustomize repository:
+
+  ```sh
+  git clone https://github.com/freeipa/freeipa-kustomize.git
+  ```
+
+- Retrieve the machine config pools by:
+
+  ```sh
+  oc get mcp
+  ```
+
+- Set the POOL environment variables with the names of the machine config pool
+  that are going to be configured:
+
+  ```sh
+  export POOL="worker"
+  ```
+
+  if you want to specify more than one:
+
+  ```sh
+  export POOL="worker master"
+  ```
+
+- Set the list of RPMs by:
+
+  ```sh
+  export RPM_PACKAGES="https://ftweedal.fedorapeople.org/runc-1.0.3-992.rhaos4.10.el8.x86_64.rpm https://ftweedal.fedorapeople.org/cri-o-1.23.0-990.rhaos4.10.git8c7713a.el8.x86_64.rpm"
```

Comment:
There is a chance that these builds will become obsolete, i.e. at a lower version than the version that ships in the cluster release, such that the rpm-ostree override will not work.  It is probably worth just mentioning that.  Also that the experimental runc build should no longer be required after v1.1 gets released and included in the cluster release.

Reply:
I have added a note

Reply:
I have updated the note again.